examples/generative/cyclegan.py

```diff
@@ -588,6 +591,8 @@ def discriminator_loss_fn(real, fake):
     generator_G=gen_G, generator_F=gen_F, discriminator_X=disc_X, discriminator_Y=disc_Y
 )
 
+cycle_gan_model.compute_output_shape(input_shape=(None, 256, 256, 3))
```

Comment:
Why this line?

Reply:
Without this line we get this error.
I followed the error message and tried to specify the input shape using the build method and got the same error.


>ValueError                                Traceback (most recent call last)
><ipython-input-11-1726c533b6c0> in <module>()
>     42     tf.data.Dataset.zip((train_horses, train_zebras)),
>     43     epochs=1,
>---> 44     callbacks=[plotter, model_checkpoint_callback],
>     45 )
>
>
>
>1 frames
>
>/usr/local/lib/python3.7/dist-packages/keras/utils/traceback_utils.py in error_handler(*args, **kwargs)
>     65     except Exception as e:  # pylint: disable=broad-except
>     66       filtered_tb = _process_traceback_frames(e.__traceback__)
>---> 67       raise e.with_traceback(filtered_tb) from None
>     68     finally:
>     69       del filtered_tb
>
>/usr/local/lib/python3.7/dist-packages/keras/saving/saving_utils.py in raise_model_input_error(model)
>     92   # If the model is not a `Sequential`, it is intended to be a subclassed model.
>     93   raise ValueError(
>---> 94       f'Model {model} cannot be saved either because the input shape is not '
>     95       'available or because the forward pass of the model is not defined.'
>     96       'To define a forward pass, please override `Model.call()`. To specify '
>
>ValueError: Model <__main__.CycleGan object at 0x7f3e527afb10> cannot be saved either because the input shape is not available or because the forward pass of the model is not defined.To define a forward pass, please override `Model.call()`. To specify an input shape, either call `build(input_shape)` directly, or call the model on actual data using `Model()`, `Model.fit()`, or `Model.predict()`. If you have a custom training step, please make sure to invoke the forward pass in train step through `Model.__call__`, i.e. `model(inputs)`, as opposed to `model.call()`.

The only way to fix it was to add this line.

Reply:
What version of TF are you using?

You don't need to add this line then. Those who want to save the model can add the line (but this example doesn't save the model and thus doesn't need it) -- and the error message tells them exactly what they need to do.

Reply:
The versions used were 2.8.2 and 2.9.2.
I have removed this line.